src/machine/CoffeeMachineSupplies.java

```diff
@@ -0,0 +1,82 @@
+package machine;
+
+public class CoffeeMachineSupplies {
+    private int amountOfWater;
+    private int amountOfMilk;
+    private int amountOfCoffeeBeans;
+    private int amountOfDisposableCups;
+
+    CoffeeMachineSupplies(int amountOfWater, int amountOfMilk, int amountOfCoffeeBeans, int amountOfDisposableCups) {
+        this.amountOfWater = amountOfWater;
+        this.amountOfMilk = amountOfMilk;
+        this.amountOfCoffeeBeans = amountOfCoffeeBeans;
+        this.amountOfDisposableCups = amountOfDisposableCups;
+    }
+
+    int getAmountOfWater() {
+        return amountOfWater;
+    }
+
+    int getAmountOfMilk() {
+        return amountOfMilk;
+    }
+
+    int getAmountOfCoffeeBeans() {
+        return amountOfCoffeeBeans;
+    }
+
+    int getAmountOfDisposableCups() {
+        return amountOfDisposableCups;
+    }
+
+    private boolean isEnoughIngredients(Coffee coffee) {
+        if (!canMake(coffee)) {
+            return false;
+        } else {
+            System.out.println("I have enough resources, making you a coffee!");
+            return true;
+        }
+    }
```

Comment:
This method should not be placed in this class, since `CoffeeMachineSupplies` doesn't know anything about coffee preparation :)

Reply:
`isEnoughIngredients(Coffee coffee)` seems like redundant, because we have `canMake(Coffee coffee)` with main logic. Only `System.out.println("I have enough resources, making you a coffee!");` is missing.